### DISCUSSION TITLE
Welcome to Apex Rollup Discussions!

### DISCUSSION THREAD

**Discussion by jamessimone** [jamessimone/apex-rollup#92] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by sfdcmx** [comment]
Hi @jamessimone hope this message finds you well!
I am was curious about the pros and cons of DLRS and Apex Rollup. I tried finding some of these here (https://www.jamessimone.net/blog/joys-of-apex/migrating-from-dlrs-with-custom-rollup/) or through this repo but could not nail down a key difference between these tools. Let me know if you have this document anywhere and if not hope you can provide some clarity on going with one or the other. Thank you!

**Reply by jamessimone** [reply]
Hey @sfdcmx, thanks for dropping on by! You're right - we need a wiki page for this. [I highlighted a response to this](https://www.reddit.com/r/salesforce/comments/silt6f/free_code_based_rollup_solution_apex_rollup/) for @pgonzaleznetwork recently on Reddit, but I can go into more detail in a dedicated Wiki page. I'll set that up immediately.

**Reply by jamessimone** [reply]
@sfdcmx I just [updated the Wiki to include a dedicate resource page for this](https://github.com/jamessimone/apex-rollup/wiki/Differences-With-DLRS) - at the moment it's essentially the content from the Reddit post, but I will expand more upon it in the near future!

**Reply by sfdcmx** [reply]
@jamessimone Thank you so much for the quick turnaround! So I did some reading in Reddit [here](https://www.reddit.com/r/salesforce/comments/silt6f/comment/hva4vev/?utm_source=share&utm_medium=web2x&context=3) and some reading re your SalesforceBen interview [here](https://www.salesforceben.com/joys-of-apex-interview-empower-your-orgs-with-apex-rollup/) and it seems like I have been away from the custom rollup game for a while.

You started Apex Rollup in Dec 2020/Jan 2021, right? I disconnected my ear from the ground regarding the custom rollup game in Oct 2020. After doing some self updating, it is fantastic to see that you are giving back to the community in this awesome way. I will dig deeper into Apex Rollup for sure and entice folks in my firm to keep a lookout for it as a scalable/LDV/more alternative to DLRS. I always knew DLRS was great, but it had some LDV limitations so I was careful with it per scenario.

Furthermore, not sure if you have seen this recording from PhillyForce about [Rollups Architecture in NPSP](https://www.youtube.com/watch?v=ArfQFffOouA&list=PLqWhVnTlsnanJnIaxKlmJKs6WdOX7L0Yc&index=2&t=14s). I was so happy to see the level of detail and enhancements in such a rollup engine, but it is only specific to NPSP data model. Take a watch and let me know what you think.

Finally, this might be a non-starter, but what are your thoughts on the level of enhancements to DLRS given that it is now part of the SFDO community? Overall it seems (without yet to get my hands dirty) Apex Rollup can be leveraged for any industry, handles LDV, can be used in Aura and LWCs and seems to handle exceptions that DLRS can't. Let me give your updated wiki a read, but glad we have connected in this discussion thread.

**Reply by jamessimone** [reply]
Apex Rollup was started in Dec 2020 and released for the first time Jan 2021, that's correct. I was not yet working for Salesforce, and thus had very little visibility into how NPSP's rollup engine might work at that time (I will have to watch that video, as I still haven't looked much at it; I will say that there are quite a few NPSP users that also have Apex Rollup installed to aid in their _own_ custom rollups).

I was happy to see Andrew hand over responsibility for DLRS to SFDO. Prior to that, he and I spoke about the potential for Apex Rollup to replace LRE engine (DLRS' rollup engine), but I felt that the surrounding architecture for DLRS would have made that an enormous project, and felt that Apex Rollup was better situated as an additional option for people looking to perform more powerful rollups. To my knowledge, the SFDO team has so far been focused more on the tooling surrounding DLRS than changes to the underlying rollup engine (which would essentially require a full re-write, but I should check back in to see if there have been any updates on that front).

In short - given the fact that Rollup Helper is essentially a paid solution, and DLRS has known limitations, I think everybody in the community is served by having another free, powerful, option, and I'm dedicated to making Apex Rollup the _best_ option for everybody. Thank you for joining the community here!
